posthog/temporal/README.md

```diff
@@ -1,20 +1,303 @@
+
+When using the Python SDK, coroutines are the recommended default for better performance, and require no extra setup. However, a specific product or feature may require executing tasks that are not well suited for `asyncio`, like a computation-heavy CPU-bound workload. For these situations, activities can be written as synchronous functions that, depending on how the Temporal workers are configured, will run in either a thread pool, or a process pool.
+
+If an activity is prone to experience temporary disruptions, it can be configured to be retried automatically by Temporal.
+
+### Temporal service
+
+The Temporal service acts as the orchestrator of all workflows and activities. This service ensures workflows are durable such that they persist even if workers crash. The service achieves this by keeping track of the execution progress of a workflow, which also enables automated retrying when any activities fail. Developers can inspect this history if required to debug their workflows.
+
+At PostHog, we rely on [Temporal Cloud](https://cloud.temporal.io) to host the Temporal service. If you require access to Temporal Cloud, you can request it via the usual help channels.
+
+### Worker
+
+Workflows and activities run in Temporal workers. The Temporal service assigns workflow and activity execution to different task queues, depending on their configuration, and workers poll these queues to execute the workflow and activity code. The workers are configured in code as they are also provided by the Temporal Python SDK.
+
+At PostHog, most of the configuration currently happens in this package, particularly in the `worker.py` module.
+
+## Implementing a product or feature with Temporal
+
+The first step when working with Temporal is deciding whether Temporal is the right tool for the job. If the application and/or feature you are implementing requires offloading work to specialized workers that must ensure durability and idempotency, offer tools to manage error handling and retry mechanisms, and you are willing to accept an increase in complexity and maintenance burden then Temporal could be the right tool for the job. Once you chosen Temporal, you will be please to know that a lot of our work here is making that increase complexity and maintenance burden small and easier to manage, so you get to re-use some of the tooling we have built when developing other products. *Yay!*
+
+Let's now dive into some of the decisions you will need to take when developing a product or feature with Temporal.
+
+### Decide which type of activity you need
+
+Temporal workers run multiple workflows and activities simultaneously. In order to enable that, we need to choose a concurrency model for our activities and, like described before, there are three types: asyncio, multithreaded, and multiprocessing. This README is not a guide on each concurrency model, so we will focus only on considerations when implementing activities using each of the concurrency models instead.
```

Comment:
Maybe worth linking to the [specific page](https://docs.temporal.io/develop/python/python-sdk-sync-vs-async) in the Temporal docs here?

Reply:
I'll add a link